docs/chap-networking.md

```diff
@@ -214,6 +214,158 @@ The Polkadot Host must actively communicate with the network in order to partici
+ -->
+The following requests must be exposed by the discovery authority mechanism to Polkadot nodes.
+
+###### Definition -def-num- Authority addresses request {#defn-auth-discovery-address-request}
+:::definition
+
+An **authority addresses request** is a request that Polkadot nodes can send to the authority discovery mechanism in order to request the addresses of an authority node. The request has the following format:
+
+$$
+\texttt{get\_addresses\_by\_authority\_id}{\left(\texttt{authorityId}\right)}
+$$
+
+**where**
+- $\texttt{authorityId}$ is the `authorityId` 256-bit identifier representing the public key of the targeted authority node. 
+
+
+**expected response** 
+
+The response to the previous query includes an enum with one of the following values:
+<!-- Option<HashSet<Multiaddr>> -->
+
+| Value                | Description                                            |
+|----------------------|--------------------------------------------------------|
+|  None                | A type representing `no value`                           | 
+|  HashSet[Multiaddr]  | An unordered collection of unique `Multiaddr` elements | 
+
+
+**with**
+- `Multiaddr` a [Multiaddr](https://github.com/libp2p/specs/blob/master/addressing/README.md#multiaddr-in-libp2p) data structure.
+  
+:::
+
+
+<!-- get_authority_ids_by_peer_id(PeerId) -->
+###### Definition -def-num- Authority identifier request {#defn-auth-discovery-auth-identifier-request}
+:::definition
+
+An **authority identifier request** is a request that polkadot nodes can send to the authority discovery mechanism in order to request the `AuthorityId` of an authority node. The request has the following format:
+
+$$
+\texttt{get\_authority\_ids\_by\_peer\_id}{\left(\texttt{PeerId}\right)}
+$$
+
+**where**
+- $\texttt{PeerId}$ is the Polkadot node’s PeerId ([Definition -def-num-ref-](chap-networking#defn-peer-id)). 
+
+
+**expected response** 
+<!-- Since we may store the mapping across several sessions, a single. `PeerId` might correspond to multiple `AuthorityId`s -->
+The response to the previous query includes an enum with one of the following values:
+<!-- Option<HashSet<AuthorityDiscoveryId> -->
+
+| Value                  | Description                                            |
+|------------------------|--------------------------------------------------------|
+|  None                  | A type representing `no value`                           | 
+|  HashSet[authorityId]  | An unordered collection of unique `authorityId` elements | 
+
+
+**with**
+- `authorityId` the 256-bit identifier representing the public key of the requested `PeerId`. 
+
+:::
+
+
+<!-- AUTHORITY DISCOVER interacting with KADEMLIA -->
+#### -sec-num- Publishing local addresses {#sect-auth-discovery-worker-publishing}
+<!-- worker component: network.put_value( hash(authorityId) , SignedAuthorityRecord);  -->
```

Comment:
You need to mention which hash algorithm and what an `authorityId` means.

Reply:
It is mentioned in the Definition